org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java

```diff
@@ -368,7 +368,7 @@ public Iterable<Pair<IStorage, IProject>> getStorages(/* @NonNull */ URI uri) {
 		// Pessimistic copy of the cachedPackageFragmentRootData
 		Collection<PackageFragmentRootData> packageFragmentRootDatas;
 		synchronized(cachedPackageFragmentRootData) {
-			packageFragmentRootDatas = cachedPackageFragmentRootData.values();
+			packageFragmentRootDatas = newArrayList(cachedPackageFragmentRootData.values());
```

Comment:
@szarnekow do you know why this was not there in the first place

Reply:
https://github.com/eclipse/xtext-eclipse/issues/1853#issuecomment-1174731216

Reply:
A copy of the list on every read is probably not what we want to do.
Maybe use a synchronized block on the list itself after it was read from the map?

Reply:
hmmmm ok. will try it out as workaround in our codebase

Reply:
we still need to check if the problem happened again